src/App/Fossa/FossaAPIV1.hs

```diff
@@ -236,6 +272,85 @@ getLatestBuild apiOpts ProjectRevision{..} = fossaReq $ do
   response <- req GET (buildsEndpoint baseUrl orgId (Locator "custom" projectName (Just projectRevision))) NoReqBody jsonResponse baseOpts
   pure (responseBody response)
 
+---------- Archive build queueing. This Endpoint ensures that after an archive is uploaded, it is scanned.
+
+archiveBuildURL :: Url 'Https -> Url 'Https
+archiveBuildURL baseUrl = baseUrl /: "api" /: "components" /: "build"
+
+archiveBuildUpload ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m) =>
+  ApiOpts ->
+  ArchiveComponents ->
+  m (Maybe C.ByteString)
+archiveBuildUpload apiOpts archiveProjects = runEmpty $
+  fossaReqAllow401 $ do
+    (baseUrl, baseOpts) <- useApiOpts apiOpts
+
+    let opts = "dependency" =: True <> "rawLicenseScan" =: True
+
+    -- The response appears to either be "Created" for new builds, or an error message for existing builds.
+    -- Making the actual return value of "Created" essentially worthless.
+    resp <- context "Queuing a build for an archive project" $ 
+      req POST (archiveBuildURL baseUrl) (ReqBodyJson archiveProjects) bsResponse (baseOpts <> opts)
+    pure (responseBody resp)
+
+---------- The signed URL endpoint returns a URL endpoint that can be used to directly upload to an S3 bucket.
+
+signedURLEndpoint :: Url 'Https -> Url 'Https
+signedURLEndpoint baseUrl = baseUrl /: "api" /: "components" /: "signed_url"
+
+getSignedURL ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m) =>
+  ApiOpts ->
+  Text ->
+  Text ->
+  m SignedURL
+getSignedURL apiOpts revision packageName = fossaReq $ do
+  (baseUrl, baseOpts) <- useApiOpts apiOpts
+
+  let opts = "packageSpec" =: packageName <> "revision" =: revision
+
+  response <- context "Retrieving a signed S3 URL" $ 
+    req GET (signedURLEndpoint baseUrl) NoReqBody jsonResponse (baseOpts <> opts)
+  pure (responseBody response)
+
+---------- The archive upload function uploads the file it is given directly to the signed URL it is provided.
+
+archiveUpload ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m) =>
+  SignedURL ->
+  FilePath ->
+  m String
```

Comment:
[nit]: Should we just return `m ()`?  We aren't using this string at all, so why bother keeping it around?

Reply:
I think returning m () is valid. The only reason I could see keeping it around is if the API was modified to return actionable information. I had it this way so I could validate we got "Created", but I removed that change for the final PR.

Reply:
If we do decide to return something, we should make it a real datatype, rather than just a string.

Reply:
Agreed, for now I'm leaving it as `m ()`